### ISSUE TITLE
APC and doctrine cache are not to use - framework documentation indicate to use abandonned components with no alternative

### ISSUE THREAD

**Issue by bruno-ds** [symfony/symfony-docs#17260]
This page https://symfony.com/doc/4.4/reference/configuration/framework.html#validation
indicate:
> ## cache
> type: string
> 
> The service that is used to persist class metadata in a cache. The service has to implement the [CacheInterface](https://github.com/symfony/symfony/blob/6.1/src/Symfony/Contracts/Cache/CacheInterface.php).
> 
> Set this option to validator.mapping.cache.doctrine.apc to use the APC cache provided by the Doctrine project.

this is wrong two times:
 - `APC` is discontinued, APCu-bc provide a compatibility layer but "As of PHP 8.0.0, apcu-bc is no longer supported"  ([source](https://www.php.net/manual/en/apcu.installation.php))
 - doctrine cache is deprecated and will no longer receive bug fixes from the Doctrine Project ([source](https://github.com/doctrine/cache)) 

So the documentation should not stipulate:
> Set this option to validator.mapping.cache.doctrine.apc to use the APC cache provided by the Doctrine project.

**What the doc should stipulate?**

Honestly, I don't know (I'm in the process of migrating to php 8 and so out of apcu-bc and `validator.mapping.cache.doctrine.apc` :/)

NB: I don't know if the doc is just out of date or if symfony still uses doctrine's APC cache.

**Comment by bruno-ds** [comment]
I slept on it, and found this [upgrade note](https://github.com/symfony/symfony/blob/4.0/UPGRADE-4.0.md#frameworkbundle):
> The validator.mapping.cache.doctrine.apc service has been removed.

so the doc. indicate to use a service that do not exists since the 4.0

Maybe it is acceptable to just remove the statement:
> Set this option to `validator.mapping.cache.doctrine.apc` to use the APC cache provided by the Doctrine project.


But, as the expected values are not straightforwards, for a good enough DX, I'd expect at least on possible value.

**Comment by bruno-ds** [comment]
After further investigations, here what I conclude:
```php
// SF 4.4, line 1341, method \Symfony\Bundle\FrameworkBundle\DependencyInjection\FrameworkExtension::registerValidationConfiguration
        if (!$container->getParameter('kernel.debug')) {
            $validatorBuilder->addMethodCall('setMappingCache', [new Reference('validator.mapping.cache.adapter')]);
        }
```

```xml
        <service id="validator.mapping.cache.adapter" class="Symfony\Component\Cache\Adapter\PhpArrayAdapter">
            <factory class="Symfony\Component\Cache\Adapter\PhpArrayAdapter" method="create" />
            <argument>%validator.mapping.cache.file%</argument>
            <argument type="service" id="cache.validator" />
        </service>
```

=> the `framework.validation.cache` entry is no more read, instead, a PhpArrayAdapter store in an "OPCache" cache.
This is done automatically as soon as `%kernel.debug%` is false.
Which is fun as it cannot be controlled using the profiler :)


=> I created a PR to remove this from the doc: https://github.com/symfony/symfony-docs/pull/17286

**Comment by javiereguiluz** [comment]
Let's close as fixed thanks to #17286.
